datastore/save_test.go

```diff
@@ -288,6 +288,20 @@ func TestSaveEmptySlice(t *testing.T) {
 	}
 }
 
+type Map map[int]int
```

Comment:
These can go inside the test

Reply:
My concern here is methods - nested method declarations are not allowed in Go (at least I never saw anything like this). It doesn't allow me to save the file with this types declared inside the test.

Reply:
Ah yeah of course, sorry I spaced on that. Just add a comment indicating the purpose/what test it is for.